File: script/mono_display.py
#!/usr/bin/env python
import sys
import roslib
import cv2
import rospy
from std_msgs.msg import String
from sensor_msgs.msg import Image
from ros_img.srv import return_data, return_dataResponse
from cv_bridge import CvBridge, CvBridgeError
from image_process import ImageProcess

class MonoDisplay(ImageProcess):
    def __init__(self,
                 name=None,
                 rate=30,
                 delta_t_buffer_size=1000):

        if name is None:
            name='mono_display'

        super(MonoDisplay, self).__init__(name=name,
                                          rate=rate,
                                          delta_t_buffer_size=delta_t_buffer_size)

    # ----------------------------------------------------------------------------------------
    # Main Loop        
    def main_process(self):
        self.output_frame = self.input_frame.copy()
        # **************************
        cv2.imshow('mono_display', self.input_frame)
        cv2.waitKey(5)

# ======================================================================================================================
def mono_display():
    mono_display = MonoDisplay()

    if len(sys.argv)==3:      
        if sys.argv[1] == '-input':
            mono_display.input_frame_subscriber_init(rostopic_name=sys.argv[2])
        else:
            mono_display.input_frame_subscriber_init()
    else:
        mono_display.input_frame_subscriber_init()
   
    mono_display.delta_t_service_init()
    mono_display.main_loop()

if __name__ == '__main__':
    try:
        mono_display()
    except rospy.ROSInterruptException:
        pass

File: src/image_process/mono_display.cpp
#include "mono_display/MonoDisplay.h"

// ======================================================================================================================
int main(int argc, char** argv)
{
	std::vector<std::string> _argv;

	for(int i=0; i<argc; i++)
	{
		_argv.push_back(argv[i]);
	}

	std::string name = "mono_display";

	ros::init(argc, argv, name);
	MonoDisplay mono_display(argc, argv, name);
	
    if(argc==3)
    {
        if(_argv[1].compare("-input")==0)
        {
            ROS_INFO("1");
            mono_display.input_frame_subscriber_init(argv[2]);
            mono_display.output_frame_publisher_init("None");
        }
        else if(_argv[1].compare("-output")==0)
        {
            ROS_INFO("2");
            mono_display.output_frame_publisher_init(argv[2]);
            mono_display.input_frame_subscriber_init("None");
        }
        else
        {
            ROS_INFO("3");
            mono_display.input_frame_subscriber_init("None");
            mono_display.output_frame_publisher_init("None");
		}
	}    
    else if(argc==5)
    {      
        if(_argv[1].compare("-input")==0)
        {
            ROS_INFO("4");
            mono_display.input_frame_subscriber_init(argv[2]);
            mono_display.output_frame_publisher_init(argv[4]);
        }
        else if(_argv[1].compare("-output")==0)
        {
            ROS_INFO("5");
            mono_display.output_frame_publisher_init(argv[2]);
            mono_display.input_frame_subscriber_init(argv[4]);
        }
        else
        {
            ROS_INFO("6");
            mono_display.input_frame_subscriber_init("None");
            mono_display.output_frame_publisher_init("None");
        }
    }
    else
    {
        ROS_INFO("7");
        mono_display.input_frame_subscriber_init("None");
        mono_display.output_frame_publisher_init("None");
    }

    mono_display.delta_t_service_init();
    mono_display.main_loop();
    cv::destroyWindow(name);
	return 0;
}